src/comm/AcceptLimiter.cc

```diff
@@ -26,37 +26,32 @@ Comm::AcceptLimiter::defer(const Comm::TcpAcceptor::Pointer &afd)
 {
     ++ (afd->isLimited);
     debugs(5, 5, afd->conn << " x" << afd->isLimited);
-    deferred_.push_back(afd);
+    deferred_.insert(afd);
 }
 
 void
 Comm::AcceptLimiter::removeDead(const Comm::TcpAcceptor::Pointer &afd)
 {
-    uint64_t abandonedClients = 0;
-    for (unsigned int i = 0; i < deferred_.size() && afd->isLimited > 0; ++i) {
-        if (deferred_[i] == afd) {
-            -- deferred_[i]->isLimited;
-            deferred_[i] = NULL; // fast. kick() will skip empty entries later.
-            debugs(5, 5, afd->conn << " x" << afd->isLimited);
-            ++abandonedClients;
-        }
+    std::set<TcpAcceptor::Pointer>::iterator it;
+    it = deferred_.find(afd);
```

Comment:
The 'it' local can have type const auto if it works, otherwise just auto and avoid the C-style pre-define.

Reply:
Changed both to "const auto"